### ISSUE TITLE
Highlight.js and <details>

### ISSUE THREAD

**Issue by AlexDawsonUK** [speced/respec#4695]
## Description of problem

When using progressive disclosure using the details element in a ReSpec document, Highlight.js ceases to function and erases the content from the rendering tree.

**URL to affected spec or repo:**

https://w3c.github.io/sustyweb/drafts/ (Note as a temporary fix we're using nohighlight as a workaround.)

**What happened (e.g., it crashed)?:**

The code inside the pre element is erased and an empty tag is rendered instead of syntax being highlighted.

**Expected behavior (e.g., it shouldn't crash):**

The syntax should be highlighted, as it is for any code that would exist outside of the details element.

**Optional, steps to reproduce:**

1.  Create a ReSpec draft document.
2. Add a **details** element with some syntax inside.
<details>
<summary>Show / Hide content.</summary>
<pre class="js"><code>
!function(e,t){"use strict";"object"==typeof module&&"object"==typeof module.exports?module.exports=e.document?t(e,!0):function(e){if(!e.document)throw new Error("jQuery requires a window with a document");return t(e)}:t(e)}("undefined"!=typeof window?window:this,function(g,e){"use strict";var t=[],r=Object.getPrototypeOf,s=t.slice,v=t.flat?function(e){return t.flat.call(e)}:function(e){return t.concat.apply([],e)},u=t.push,i=t.indexOf
</code></pre>
</details>

- This test showcases a basic ReSpec with it non-functional [test.txt](https://github.com/w3c/respec/files/15023847/test.txt)
- This test showcases a basic HTML file with it functional [test2.txt](https://github.com/w3c/respec/files/15023848/test2.txt)

**Note:** I have verified that this issue does not occur in a normal HTML document using highlight.js therefore it possibly is due to ReSpec or its implementation of highlight?

**Comment by sidvishnoi** [comment]
@AlexDawsonUK I cannot reproduce the error with `test.txt` you shared.

<img width="1552" alt="image" src="https://github.com/w3c/respec/assets/8426945/502d0bab-5bbc-4644-ad28-273a66aeff19">


Please check there's no `<` like unescaped HTML tags in your `<pre>` content. Also, please confirm if it works fine without a parent `<details>`.

**Comment by sidvishnoi** [comment]
Thanks @AlexDawsonUK. I'll check in Chrome too. Could possibly be a ReSpec issue.

**Comment by sidvishnoi** [comment]
Doesn't happen with `<details open>` in Chrome 🤓

**Comment by AlexDawsonUK** [comment]
Yes, I noted that odd quirk of behavior, and I even tried writing some JavaScript to close all the details components once everything had finished loading (hoping that it would give Highlight enough time todo its thing and then close them until required). But alas, whatever is causing the bug wouldn't have any of it. 🤷🏻‍♂️

I've expanded the testing and can confirm that all chromium browsers are affected (I tested in Chrome / Edge / Brave).
Firefox and Safari are both free from the issue though so I guess we can blame Google for a weird implementation? 😈

**Comment by sidvishnoi** [comment]
Agrees on blaming Blink behavior, as Safari and Firefox work fine. The actual issue is Blink returns empty string with `innerText` when content is not visible. I think ReSpec could use textContent, but I'm concerned about breaking whitespace.
https://github.com/w3c/respec/blob/fca26bdd681a2f3ecc3ff769375f09472fed439b/src/core/highlight.js#L26

cc @marcoscaceres Any idea why did we use `innerText`?

**Comment by AlexDawsonUK** [comment]
**Update:** I've created a potential workaround patch that allows functioning rendering engines to continue to enjoy syntax highlighting while chromium functions with it disabled. It requires the use of the _nohighlight_ class on all affected code.

```
<script async>
	const isFirefox = navigator.userAgent.toLowerCase().includes('firefox');
	const isSafari = navigator.vendor && navigator.vendor.indexOf('Apple') > -1 &&
             navigator.userAgent &&
             navigator.userAgent.indexOf('CriOS') == -1 &&
             navigator.userAgent.indexOf('FxiOS') == -1;
	if (isFirefox || isSafari) {
		for (let i = 0; i < document.getElementsByTagName('pre').length; i++) {
			if (document.getElementsByTagName('pre')[i].className = "nohighlight" ) {
				document.getElementsByTagName('pre')[i].className = ""
			}
		}
	}
</script>
```
It requires highlighting to be enabled on functional browsers rather than disabling it on non-functional ones due to the issue of the loss of content being displayed from the rendering tree as mentioned previously regarding innerText.

**Comment by marcoscaceres** [comment]
I can't recall, but probably because `.textContent` is wonky. We could try it though.

**Comment by nigelmegitt** [comment]
> `if (document.getElementsByTagName('pre')[i].className = "nohighlight" )`
> `	document.getElementsByTagName('pre')[i].className = ""`

You probably want:

`document.getElementsByTagName('pre')[i].classList.remove("nohighlight");`
